src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -483,21 +488,55 @@ public void generateComponentLocationAnalysisIfEnabled(BdioResult bdio) throws O
      * @param rapidFullResults
      * @throws OperationException
      */
-    public void generateComponentLocationAnalysisIfEnabled(List<DeveloperScansScanView> rapidFullResults) throws OperationException {
+    public void generateComponentLocationAnalysisIfEnabled(List<DeveloperScansScanView> rapidFullResults, BdioResult bdio) throws OperationException {
         if (detectConfigurationFactory.isComponentLocationAnalysisEnabled()) {
-            if (!rapidFullResults.isEmpty()) {
-                auditLog.namedPublic(
-                        "Generate Component Location Analysis File for Reported Components",
-                        () -> (new GenerateComponentLocationAnalysisOperation()).locateComponentsForNonPersistentOnlineDetectorScan(rapidFullResults, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
-                );
-            } else {
+            if (applicableDetectorsIncludeAtLeastOneSupportedDetector(bdio.getApplicableDetectorTypes())) {
+                if (!rapidFullResults.isEmpty()) {
+                    auditLog.namedPublic(
+                            "Generate Component Location Analysis File for Reported Components",
+                            () -> (new GenerateComponentLocationAnalysisOperation()).locateComponentsForNonPersistentOnlineDetectorScan(rapidFullResults, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
+                    );
+                } else {
+                    logger.info(ReportConstants.RUN_SEPARATOR);
+                    logger.info("Component Location Analysis requires non-empty Rapid Scan results. Skipping location analysis.");
+                    logger.info(ReportConstants.RUN_SEPARATOR);
+                }
+        } else {
                 logger.info(ReportConstants.RUN_SEPARATOR);
-                logger.info("Component Location Analysis requires non-empty Rapid Scan results. Skipping location analysis.");
+                logger.info("Component Location Analysis only supports NPM, Maven, Gradle and NuGet detectors.");
```

Comment:
Nit: Component Location Analysis supports NPM, Maven, Gradle and NuGet detectors **only**.